schema-registry/serdes/src/main/java/com/hortonworks/registries/schemaregistry/serdes/avro/AbstractAvroSerDesProtocolHandler.java

```diff
@@ -42,13 +42,13 @@
      */
     public static final String READER_SCHEMA = "reader.schema";
 
-    private final DefaultAvroSerDesHandler defaultAvroSerDesHandler;
+    private final AvroSerDesHandler avroSerDesHandler;
 
     protected final Byte protocolId;
 
-    protected AbstractAvroSerDesProtocolHandler(Byte protocolId) {
+    protected AbstractAvroSerDesProtocolHandler(Byte protocolId, AvroSerDesHandler avroSerDesHandler) {
```

Comment:
`avroSerDesHandler ` can be removed and this abstract class uses `DefaultAvroSerDesHandler ` but the respective ser/des payload methods should be over ridden by implementors instead of passing `AvroSerDesHandler` instance.